src/shared/database/index.ts

```diff
@@ -447,13 +449,13 @@ export async function signingToMaster(
  * @returns Hourly Agreement for validator between start and end.
  */
 async function getHourlyAgreementScores(
-  validator: string,
+  validator: ValidatorKeys,
   start: Date,
   end: Date,
 ): Promise<AgreementScore[]> {
   return query('hourly_agreement')
     .select(['agreement'])
-    .where({ master_key: validator })
+    .where({ main_key: validator.master_key ?? validator.signing_key })
```

Comment:
If you make `ValidatorKeys` a class you can add a property `ValidatorKeys.main_key` which does this calculation on each call to avoid repeating this check since you do it fairly often.

Reply:
I think it's more readable this way, and it feels a bit overkill to have a class just for that in JS land

Reply:
Fair enough :)